easybuild/easyblocks/g/gcc.py

```diff
@@ -366,6 +366,63 @@ def prep_extra_src_dirs(self, stage, target_prefix=None):
             'versions': versions
         }
 
+    def map_nvptx_capability(self):
+        """
+        Convert PTX ISA architecture passed via EasyBuild configs to a version which is understood by GCC.
+        Valid architecture strings include 'sm_30', 'sm_35', 'sm_53', 'sm_70', 'sm_75', 'sm_80'.
+        (as of GCC 14.1.0). Some additional architectures may be mapped.
+        See: https://github.com/gcc-mirror/gcc/commit/de0ef04419e90eacf0d1ddb265552a1b08c18d4b
+
+        As this list is updated regularly, try to parse the GCC source file (gcc/config/nvptx/nvptx.opt)
+        and extract the supported architectures and their mapping. Based on the result, determine the lowest
+        architecture required to support all 'cuda_compute_capabilities' and return this value.
+        """
+        # Determine which compute capabilities are configured. If there are none, return immediately.
+        cuda_cc_list = build_option('cuda_compute_capabilities') or self.cfg['cuda_compute_capabilities']
+        if cuda_cc_list is None:
+            return None
+        cuda_sm_list = [f"sm_{cc.replace('.', '')}" for cc in cuda_cc_list]
+
+        architecture_mappings_file = os.path.join(self.cfg['start_dir'], 'gcc', 'config', 'nvptx', 'nvptx.opt')
+        if not os.path.exists(architecture_mappings_file):
+            warn_msg = "Tried to parse nvptx.opt but file was not found. " \
+                       "Please check the path and update the EasyBlock if necessary!"
+            self.log.warning(warn_msg)
+            return None
+
+        # We want to read the mappings found in the GCC sources and create a map for this.
+        # We're searching for the following pattern:
+        # march-map=sm_32
+        # Target RejectNegative Alias(misa=,sm_30)
+        gcc_architecture_mappings = {}
+        with open(architecture_mappings_file) as file:
+            file_content = file.read().splitlines()
+            for line_idx in range(len(file_content)):
+                line = file_content[line_idx].strip()
+                if line.startswith("march-map="):
+                    key = line.split('=')[1]
+                    # Mapped architecture can be found in the following line
+                    line = file_content[line_idx + 1]
+                    if "misa=," not in line:
+                        warn_msg = "Tried to parse nvptx.opt but failed to extract mapped architectures! " \
+                                   "Choosing default of GCC."
+                        self.log.warning(warn_msg)
+                        # Bail out, since results of mapping cannot be trusted
+                        return None
+                    value = line.split('misa=,')[1][:-1]
```

Comment:
Maybe better to avoid surprises, only strip off trailing `)` rather than cutting off last character?

```python
value = line.split('misa=,')[1].rstrip(')')